zerver/forms.py

```diff
@@ -108,9 +108,7 @@ class HomepageForm(forms.Form):
 
     def __init__(self, *args, **kwargs):
         # type: (*Any, **Any) -> None
-        self.string_id = kwargs.get("string_id")
-        if "string_id" in kwargs:
-            del kwargs["string_id"]
+        self.string_id = kwargs.pop("string_id", None)
```

Comment:
I would argue the previous code was a bit more readable, but this change is fine.